AFQ/tasks/tractography.py

```diff
@@ -348,3 +458,18 @@ def get_tractography_plan(kwargs):
                 seed_mask.get_image_getter("tractography")))
 
     return pimms.plan(**tractography_tasks)
+
+
+def _gen_seeds(n_seeds, params_file, seed_mask=None, seed_threshold=0,
+               thresholds_as_percentages=False,
+               random_seeds=False, rng_seed=None):
+    if isinstance(params_file, str):
+        params_img = nib.load(params_file)
+    else:
+        params_img = params_file
+
+    affine = params_img.affine
```

Comment:
For this, I would just ask for affine for _gen_seeds instead of params_file. Then, use data_imap['dwi_affine'] to get the affine. It will be the same as the params_img affine, and then you don't have to load the params_img just for the affine

Reply:
In fact, you probably don't need this function. Instead of:
```
                    seeds = _gen_seeds(this_tracking_params['n_seeds'],
                                       params_file, **this_tracking_params)
```
do:
```
                   seeds = gen_seeds(
                       this_tracking_params['seed_mask'],
                       this_tracking_params['seed_threshold'],
                       this_tracking_params['n_seeds'],
                       this_tracking_params['thresholds_as_percentages'],
                       this_tracking_params['random_seeds'],
                       data_imap["dwi_affine"])
```
I guess this is a little longer which is part of what you are avoiding with the keyword arguments, but I think it's the way to go here.

Reply:
Another approach would be to have `track` and `gen_seeds` accept `tracking_params` outright instead of accepting each of these arguments individually. You could even then have tracking params be a class with documentation elsewhere, like:
```
class TrackingParams:
    seed_mask = None
    seed_threshold = 0
    ...
```

Reply:
Some of these arguments are optional, would doing this not override the default arguments with None? For example thresholds_as_percentages would be set to None instead of False here?

Reply:
No, when your in `streamlines`, your in the task system that gets called from either GroupAFQ or ParticipantAFQ. In here, we try to make sure that everything is set to a reasonable default, sometimes based on other options people set in those objects. In particular, these lines 257-281: https://github.com/yeatmanlab/pyAFQ/blob/6d0ffb699b975dde309dc8ada52e7b70976325f4/AFQ/tasks/tractography.py#L257
make sure that thresholds_as_percentages will either be directly set by the user whatever the default is in `track`

Reply:
Got it! Thanks for your help, the new commit implements your advice.